File: apps/frontend/src/model/GameState.ts
export interface Player {
  name: string;
  character: string;
  current_room: string;
  has_made_suggestion: boolean;
}

export interface GameState {
  current_turn: string;
  players: Player[];
  rooms: string[];
  suggestions: { player: string; suggestion: string }[];
}

File: apps/backend/model/GameState.py
from pydantic import BaseModel
from typing import List, Optional

class Player(BaseModel):
    name: str
    character: str
    current_room: str
    has_made_suggestion: bool

class GameState(BaseModel):
    current_turn: str
    players: List[Player]
    current_turn: int
    game_over: bool
    winner: Optional[str] = None
